runtime/compiler/control/CompilationThread.cpp

```diff
@@ -247,7 +257,124 @@ jitSignalHandler(struct J9PortLibrary *portLibrary, U_32 gpType, void *gpInfo, v
 
    return J9PORT_SIG_EXCEPTION_CONTINUE_SEARCH;
    }
-
+#ifdef COMPRESS_AOT_DATA
+#ifdef J9ZOS390
+#pragma convlit(suspend)
+#endif
+/*
+ * \brief
+ *    Inflates the data buffer using zlib into output buffer
+ *
+ * \param 
+ *    buffer Input buffer that is going to be inflated 
+ * 
+ * \param 
+ *    numberOfBytes Size of the data in the input buffer to inflate
+ * 
+ * \param
+ *    outBuffer Output buffer to hold the deflated data
+ * 
+ * \param
+ *    unCompressedSize Size of the original data
+ * 
+ * \return
+ *    Returns DECOMPRESSION_FAILED if it can not inflate the buffer
+ * 
+ */
+static
+int inflateBuffer(U_8 *buffer, int numberOfBytes, U_8 *outBuffer, int unCompressedSize)
+   {
+   Bytef *in = (Bytef*)buffer;
+   Bytef *out = (Bytef*)outBuffer;
+   z_stream _stream;
+   _stream.zalloc = Z_NULL;
+   _stream.zfree = Z_NULL;
+   _stream.opaque = Z_NULL;
+   _stream.avail_in = 0;
+   _stream.next_in = Z_NULL;
+   int ret = inflateInit(&_stream);
+   if (ret != Z_OK)
+      return DECOMPRESSION_FAILED;
+   _stream.avail_out = unCompressedSize;
+   _stream.next_out = out;
+   _stream.next_in = in;
+   _stream.avail_in = numberOfBytes;
+   ret = inflate(&_stream, Z_NO_FLUSH);
+   switch (ret)
+      {
+      case Z_NEED_DICT:
+      case Z_DATA_ERROR:
+      case Z_MEM_ERROR:
+         (void)inflateEnd(&_stream);
+         return DECOMPRESSION_FAILED;
+      }
+   ret = 0;
+   if (_stream.avail_out != 0)
+      {
+      TR_ASSERT(0, "Fails while inflating buffer");
+      ret = DECOMPRESSION_FAILED;
+      }
+   (void)inflateEnd(&_stream);
+   return ret;
+   }
+
+/*
+ * \brief
+ *    Deflates the data buffer using zlib into output buffer
+ *
+ * \param 
+ *    buffer Input buffer that is going to be deflated 
+ * 
+ * \param 
+ *    numberOfBytes Size of the data in the input buffer to deflate
+ * 
+ * \param
+ *    outBuffer Output buffer to hold the deflated data
+ * 
+ * \param
+ *    level Level of compression
+ * 
+ * \return
+ *    Size of data in the deflated bufer
+ * 
+ */
+static
+int deflateBuffer(const U_8 *buffer, int numberOfBytes, U_8 *outBuffer, int level)
+   {
+   z_stream _stream;
+   _stream.zalloc = Z_NULL;
+   _stream.zfree = Z_NULL;
+   _stream.opaque = Z_NULL;
+   int ret=deflateInit(&_stream, level);
+   if (ret != Z_OK)
+      return COMPRESSION_FAILED;
+   // Start deflating
+   _stream.avail_in = numberOfBytes;
+   _stream.next_in = (Bytef*) buffer;
+   _stream.avail_out = numberOfBytes;
+   _stream.next_out = (Bytef*) outBuffer;
+   ret = deflate(&_stream, Z_FINISH);
+   /**
+    * ZLIB return Z_STREAM_END if the buffer streamed to be deflated is finished.
+    * That Return COMPRESSION_FAILED in case we are ending up inflating the data.@brief 
```

Comment:
I'm guessing the `@brief` isn't supposed to be at the end of the line?

Reply:
Yeah, it got copied over when I changed comment format